lib/awful/completion.lua

```diff
@@ -153,7 +153,7 @@ function completion.shell(command, cur_pos, ncomp, shell)
     end
 
     local str = command:sub(1, cword_start - 1) .. output[ncomp] .. command:sub(cword_end)
-    cur_pos = cword_end + #output[ncomp] + 1
+    cur_pos = cword_start + #output[ncomp]
```

Comment:
Perhaps some words on the commit message on what is going on here? This will now return a smaller number than previously, I think. Why?